### ISSUE TITLE
Crash when closing terminals

### ISSUE THREAD

**Issue by phw** [gnunn1/terminix#302]
When I load a session from file and start closing terminals in this session Terminix will eventuall crash with:

```
core.exception.RangeError@source/gx/terminix/session.d(326): Range violation
----------------
??:? _d_arraybounds [0xbc2a73]
??:? gx.terminix.session.__array [0x96efb7]
source/gx/terminix/session.d:326 void gx.terminix.session.Session.removeTerminal(gx.terminix.terminal.terminal.Terminal) [0x969ccc]
source/gx/terminix/session.d:528 void gx.terminix.session.Session.closeTerminal(gx.terminix.terminal.terminal.Terminal) [0x96a82c]
source/gx/terminix/session.d:536 void gx.terminix.session.Session.onTerminalClose(gx.terminix.terminal.terminal.Terminal) [0x96a85c]
source/gx/terminix/terminal/terminal.d:888 void gx.terminix.terminal.terminal.Terminal.notifyTerminalClose() [0x9775af]
source/gx/terminix/terminal/terminal.d:544 _D2gx8terminix8terminal8terminal8Terminal13createActionsMFC3gio17SimpleActionGroup17SimpleActionGroupZ76__T13__dgliteral17TC4glib7Variant7VariantTC3gio12SimpleAction12SimpleActionZ13__dgliteral17MFC4glib7Variant7VariantC3gio12SimpleAction12SimpleActionZv [0x97530e]
../../.dub/packages/gtk-d-3.3.1/src/gio/SimpleAction.d:243 extern (C) void gio.SimpleAction.SimpleAction.callBackActivate(gtkc.giotypes.GSimpleAction*, gtkc.glibtypes.GVariant*, gio.SimpleAction.SimpleAction) [0x9c7818]
??:? g_closure_invoke [0x8590dfa4]
```

This does only happen if loaded from file. If I recreate the session within terminal by splitting closing the terminals works as expected. To reproduce try to load e.g. [Standard.json](https://github.com/gnunn1/terminix/files/251842/Standard.json.txt) and start closing terminals using the keyboard shortcut.

Interestingly I cannot reproduce this reliable when using the close button, but when using the keyboard shortcut it will crash always on the first or second terminal I close.

**Comment by gnunn1** [comment]
I can't reproduce, can you post your session json file please. Also, are you loading the file from the command line or via the open menu?

**Comment by gnunn1** [comment]
Oops, I missed the bit about shortcuts, let me try again. Judging my the line number looks related to the mru focus tracking that was recently added.

**Comment by phw** [comment]
Confirm fixed, thanks
